### PR TITLE
Merge inner_silence_uat into master

### PR DESCRIPTION
**Pull request by cristinamghita**

### CODE REVIEW

File: backend/de.metas.acct.base/src/main/java-legacy/org/compiere/acct/FactLine.java [hunk 1]
```diff
@@ -741,6 +750,15 @@ private static Optional<Instant> computeAcctSchemaElementAsDate(
 				.map(date -> date.toInstant(doc.getServices()::getTimeZone));
 	}
 
+	private Optional<YearAndCalendarId> computeYearAndCalendarId(final AcctSchema acctSchema, final Doc<?> m_doc, final DocLine<?> m_docLine)
```
Comment (teosarca) [inline backend/de.metas.acct.base/src/main/java-legacy/org/compiere/acct/FactLine.java:753]: static

File: backend/de.metas.acct.base/src/main/java-legacy/org/compiere/acct/FactLine.java [hunk 2]
```diff
@@ -741,6 +750,15 @@ private static Optional<Instant> computeAcctSchemaElementAsDate(
...
+	{
+		int calendarId = computeAcctSchemaElementAsId(AcctSchemaElementType.HarvestingCalendar, acctSchema, m_doc, m_docLine).orElse(0);
+		int yearId = computeAcctSchemaElementAsId(AcctSchemaElementType.HarvestingYear, acctSchema, m_doc, m_docLine).orElse(0);
+
+		return calendarId > 0 && yearId >0 ? Optional.of(YearAndCalendarId.ofRepoId(yearId, calendarId)) : Optional.empty();
```
Comment (teosarca) [inline backend/de.metas.acct.base/src/main/java-legacy/org/compiere/acct/FactLine.java:758]: consider introducing YearAndCalendarId.optionalOfRepoId

File: backend/de.metas.acct.base/src/main/java-legacy/org/compiere/acct/FactLine.java [hunk 3]
```diff
@@ -1469,6 +1492,18 @@ public void updateFromDimension(final AccountDimension dim)
 		{
 			setUserElementDate2(dim.getUserElementDate2());
 		}
+
+		if (dim.isSegmentValueSet(AcctSegmentType.HarvestingCalendar) && dim.isSegmentValueSet(AcctSegmentType.HarvestingYear))
+		{
+			int calendarId = dim.getC_Harvesting_Calendar_ID();
```
Comment (teosarca) [inline backend/de.metas.acct.base/src/main/java-legacy/org/compiere/acct/FactLine.java:1498]: consider using YearAndCalendarId.optionalOfRepoId(....).ifPresent(....)

File: backend/de.metas.acct.base/src/main/java-legacy/org/compiere/acct/FactLineBuilder.java [hunk 4]
```diff
@@ -3,6 +3,10 @@
 import de.metas.acct.Account;
 import de.metas.acct.AccountConceptualName;
 import de.metas.acct.api.impl.ElementValueId;
+import de.metas.acct.api.AcctSchema;
+import de.metas.acct.api.AcctSchemaElement;
+import de.metas.acct.api.AcctSchemaElementType;
+import de.metas.acct.api.PostingType;
```
Comment (teosarca) [inline backend/de.metas.acct.base/src/main/java-legacy/org/compiere/acct/FactLineBuilder.java:9]: consider organizing imports

**Review by teosarca — APPROVED** [review]

**Comment by cristinamghita** [comment]
The only cucumber failed is the one that is already failling in from master:
image
Merging it!

**Pull request merged**